typing/typetexp.ml

```diff
@@ -298,7 +282,7 @@ and transl_type_aux env policy styp =
           in
           let path, decl = Env.find_type_by_name lid2 env in
           ignore(Env.lookup_cltype ~loc:lid.loc lid.txt env);
-          (path, decl, false)
+          (path, decl)
         with Not_found ->
           ignore (Env.lookup_cltype ~loc:lid.loc lid.txt env); assert false
       in
```

Comment:
I am kind of of wondering whether it would make sense to replace all this by a simple call to `lookup_type ~loc:lid.loc lid2 env`

But this may go in a separate PR.
          let lid2 =
             match lid.txt with
               Longident.Lident s     -> Longident.Lident ("#" ^ s)
            | Longident.Ldot(r, s)   -> Longident.Ldot (r, "#" ^ s)
            | Longident.Lapply(_, _) -> fatal_error "Typetexp.transl_type"
           in
           try Env.find_type_by_name lid2 env with Not_found -> assert false

Reply:
I think separate PR might be better, to keep this PR deletion-only.